youtube_dl/utils.py

```diff
@@ -1835,6 +1836,17 @@ def write_json_file(obj, fn):
                 os.unlink(fn)
             except OSError:
                 pass
+        mask = os.umask(0)
+        os.umask(mask)
+        os.chmod(tf.name,
+                 (stat.S_IRUSR
+                  | stat.S_IWUSR
+                  | stat.S_IRGRP
+                  | stat.S_IWGRP
+                  | stat.S_IROTH
+                  | stat.S_IWOTH)
+                 & ~mask)
```

Comment:
This must be in a separate try-except so that it does not break renaming to correct name if any of these statements fails.